apps/veil/src/pages/portfolio/ui/shield-unshield.tsx

```diff
@@ -0,0 +1,228 @@
+import type { UnifiedAsset } from '@/pages/portfolio/api/use-unified-assets.ts';
+import { Button } from '@penumbra-zone/ui/Button';
+import { Tooltip } from '@penumbra-zone/ui/Tooltip';
+import { Widget } from '@skip-go/widget';
+import { useEffect, useState, useMemo } from 'react';
+import { X } from 'lucide-react';
+import { useRegistry } from '@/shared/api/registry.ts';
+import { getMetadata } from '@penumbra-zone/getters/value-view';
+
+/**
+ * Computes the IBC denom hash for Penumbra-1
+ * If the denom already starts with "ibc/", it's assumed to be already computed.
+ * If the denom starts with "transfer/", it's assumed to be a full trace path.
+ * Otherwise, constructs the trace path using channelId and denom.
+ */
+async function computeIbcDenom(denom: string, channelId: string): Promise<string> {
```

Comment:
question: i understand that this function hashes the denom with SHA-256 and that this hash later goes to the Skip widget as `defaultRoute.destAssetDenom` prop. Is it the recommended and required way to do so?

Reply:
It is required if we want to prefill the route in Skip, didn't find a way to do this better - if you or anyone else has improvements, i'm all for them.

Reply:
don't mind if it's required, could be nice to add a comment with a link to the docs recommending it

Reply:
Done